omdb-client/impl/src/main/java/com/garret/movies/omdb/impl/client/OmdbClientImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.garret.movies.omdb.impl.client;
+
+import com.garret.movies.dao.entity.Movie;
+import com.garret.movies.omdb.api.OmdbClient;
+import com.garret.movies.omdb.impl.exception.OmdbApiException;
+import com.garret.movies.omdb.impl.util.MovieUtil;
+import com.omertron.omdbapi.OMDBException;
+import com.omertron.omdbapi.OmdbApi;
+import com.omertron.omdbapi.model.OmdbVideoBasic;
+import com.omertron.omdbapi.model.OmdbVideoFull;
+import com.omertron.omdbapi.tools.OmdbParameters;
+import com.omertron.omdbapi.tools.Param;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.ListUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class OmdbClientImpl implements OmdbClient {
+
+    private OmdbApi omdbApi;
```

Comment:
@Garret-vin Add final and use @RequiredArgsConstuctor. Otherwise if you will add some class variable for internal use it will be added to constructor as well which you don't need